src/relations.jl

```diff
@@ -1212,6 +1239,44 @@ PhasePartition_equil(ts::AbstractPhaseNonEquil) = PhasePartition_equil(
     typeof(ts),
 )
 
+
+"""
+    PhasePartition_equil_given_p(param_set, T, p, q_tot, phase_type)
+Partition the phases in equilibrium, returning a [`PhasePartition`](@ref) object using the
+[`liquid_fraction`](@ref) function where
+ - `param_set` an `AbstractParameterSet`, see the [`Thermodynamics`](@ref) for more details
+ - `T` temperature
+ - `p` air pressure
+ - `q_tot` total specific humidity
+ - `phase_type` a thermodynamic state type
+The residual `q.tot - q.liq - q.ice` is the vapor specific humidity.
+"""
+function PhasePartition_equil_given_p(
+    param_set::APS,
+    T::FT,
+    p::FT,
+    q_tot::FT,
+    phase_type::Type{<:ThermodynamicState},
+) where {FT <: Real}
+
+    q_v_sat = q_vap_saturation_from_pressure(param_set, q_tot, p, T, Liquid())
+    _liquid_frac = liquid_fraction(param_set, T, phase_type)                    # fraction of condensate that is liquid
+    q_c = q_tot - q_v_sat
```

Comment:
Since you are computing q_c from this q_v_sat, I believe that the p_vap_saturation has to be computed using the weighted average of latent heats, with the constructor 

`saturation_vapor_pressure(
    param_set::APS,
    phase_type::Type{<:ThermodynamicState},
    T::FT,
)`

Right now you are computing the q_vap_sat for liquid.

Reply:
great catch. I guess that what you mean is that 'saturation_vapor_pressure' used inside 'q_vap_saturation_from_pressure' should be changed here? 
https://github.com/CliMA/Thermodynamics.jl/blob/7a560298c1f49288ad30b74b54c2a8c5392e0f8d/src/relations.jl#L1040